src/components/nav-icon-item/nav-icon-item.tsx

```diff
@@ -0,0 +1,63 @@
+import { AnchorHTMLAttributes, ButtonHTMLAttributes, FC, MouseEventHandler, ReactNode } from 'react'
+import { ElAnchorNavIconItem, ElButtonNavIconItem } from './styles'
+
+interface CommonNavIconItemProps {
+  /**
+   * The JSX element to render as the icon
+   */
+  icon: ReactNode
+
+  /**
+   * Whether the nav item is active
+   * @default false
+   **/
+  isActive?: boolean
+}
+
+interface NavIconItemAsButtonElementProps extends CommonNavIconItemProps, ButtonHTMLAttributes<HTMLButtonElement> {
+  href?: never
+  target?: never
```

Comment:
**question:** is this to try and get a proper discriminated union happening? Just seems odd that we're explicitly excluding props that, at face value, wouldn't be part of this type anyway (i.e. `ButtonHTMLAttributes<HTMLButtonElement>` doesn't define either of these, right?)

Reply:
> is this to try and get a proper discriminated union happening?

yes, for discriminated union types. Unfortunately, neither `ButtonHTMLAttributes` from React nor `HTMLButtonElement` supports this

Reply:
Okay, and does this actually work the way you intend? I'm having a hard time visualising what the impact of not having this defined would be 😵  Maybe you could provide a [TS Playground](https://www.typescriptlang.org/play/) link demonstrating how defining these two as `never` helps with the discriminated union type?

Reply:
> Maybe you could provide a [TS Playground](https://www.typescriptlang.org/play/) link demonstrating how defining these two as never helps with the discriminated union type?

I left a simple demonstration in [here](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAQgrsYB7AdgeRQYQDYEsDGA1lALxQAUAlKQHxQDkAzgBZJzYAmUAThMHNxRQABgCMEyFMKiNg3XCgDm9AFChIUAIIp8rbhhwFiZKrQYs2nHnwFDhAQx17ps+UtUqFwCNwBm9-Gh4RFQABW4kMEYoAG8VKChmXl8AfgAuKBQIADcfAG54qFRDIgzgyQM8IgKAXxVPFG8-AOhtXSRucMjouISkiFSM1wVFAoTiqsIMtr1Ko1r69WgASW8AW1JYCTCIqKgAHy0nDq6ogpV8VFkoXEZy1E3yXHWM1Yg1ygzn95vo+5RTtESHQAITfNYAOn6vnOlxQ10UfAAsnx7Bx7MB7I9wa91tRgbFCrhfBRbv8nnjqL0ElA4dcYkUsJMADSJZJQGqbcFjGl0pDYCAQ7BIRTkej-eiUQoJPkCoUi8jQyh5KAAelViSQuW4UGAzGg0NZzygAHdLFw1vZuMR7NFHFA4CgOAMFBAONLaVd+YLhaKJkYqMq1RrWNrdfqRP6iFRhEbgDJWOwuGA3PGAEQWJPWfiCETiEJSGRyEZpwqciDYRjQakyq7xhlRwis6EcrnrHm1+He+Wi+gzDqSj2yn0KpUq9WasN6g3JOOm81QS3WqC2ldCR3O3yurgdFdFtyKIdeuW+8iNwPjkNanzh6DCc+UaTGzNWFNeKAZxNWXg5uyOdrcC4xZKKWCR1DU5wThAAAe9hrGAAoOow9iIhcdZQPmkhvGsZTbACuxAoS4xMkYGSmASTBflwP62HmeFAQe9DMioEH1HS8b-no2HTMcnQEZs1LQhk9DMIgURpOqMFwQhgqXGsTGFI2ZH4nQlELjRuYOLxDEjAprEqIiwAopi6KYuQmGoNhyoGciqKmfY5CcR0Vl5EAA)

Reply:
Thanks for the playground; [here's an updated version](JYWwDg9gTgLgBDAnmApnA3nAggOwMYAW0AEgCoCyAMljDFMAEYCuMKAzgDRwBCLMEOMlRp1GLdlwBiAYS7kITNigCiANxQ4YxAIY4AJgBsUUOAF84AMygQQcAORQU2vDDsAoNwHpPPPgKHUtPTMrGxwACIA8soAynAAcpGkcMD4Bkx6aASOFnC6egjaUADmKDAAdHAAkvB6EOxwSii2) that I think shows the minimum we need (just `href`, not all the ones you have here)